### ISSUE TITLE
osbs.wait_for_build_to_finish() doesn't wait for build to finish

### ISSUE THREAD

**Issue by pbabinca** [DBuildService/osbs#102]
`osbs.wait_for_build_to_finish()` doesn't work as advertised. `wait()` sometimes (always?) prints:
```
osbs.core:237 object has changed: 'u'MODIFIED'', status: 'Running', name: 'container-docker-20150525-142550'
osbs.core:239 matching build found
```
but this isn't the state `wait()` expects:
```
obj_status: u'running' not in states: [u'failed', u'complete', u'error', u'cancelled']
```

If the connection is closed now the error with build not found is printed.

I guess it should restart wait() in this case. Maybe some retry count or timeout needs to be implemented to not wait forever.

Note: I use #101 with unicode literals on python 2.6.

**Comment by TomasTomecek** [comment]
As I [said](https://github.com/DBuildService/osbs/pull/101#issuecomment-105241893) in #101; feel free to grab this one. I am already kind of burn out on this rhel6/pycurl/unicode-madness.

**Comment by TomasTomecek** [comment]
I thought this is about something different.

**Comment by TomasTomecek** [comment]
Fixed by #108
